### DISCUSSION TITLE
Tesla api not working

### DISCUSSION THREAD

**Discussion by fkhera** [timdorr/tesla-api#410] [Q&A]
Tesla api not working again. Any ideas ? Did something change on the site ?

It appears they activated captcha again.  Just set to 'True' in latest code here and enter captcha api key:
This might be temporary, but who knows I will keep this posted.  It does appear as of 8pm today, the code didn't work without out captcha codes, and last time this lasted about 48 hours and then they disabled captcha check.

**Comment by JonG67x** [comment]
Yes, same problem here

**Reply by JonG67x** [reply]
I've found if I display the return from the oauth2/v3/authorize call when its 200 but not a MFA, get the captcha value and then resubmit the oauth2/v3/authorize with all the same values (with the captcha value set) then it continues and either requests for the MFA values as before, or completes normally. Its made my code a bit messy but its now working again.

**Reply by dacastro4** [reply]
@JonG67x I'm having the same issue. Do you mind explaining a little bit further where are you the captcha value from?

**Reply by JonG67x** [reply]
When you do the authorize call, if you get http_code =200 its looking for further authorisation. I check for /mfa/verify in the body and if that exists it wants the MFA code (which is a different branch of logic), if it doesn't I assume its after the capture value.  I do a call to https://auth.tesla.com/captcha with all the usual values (I use Curl with CURLOPT_HEADER = 0) and take the result which is the captcha image.  I take that and embed it in a simple web page with a form to capture the result and then feed that back into the authroize call with all the same values.

**Comment by fkhera** [comment]
I like the idea of checking If captcha is present. Right now I am just
basing it off what my users tell me and using flags to turn on and off
captcha.

As far as I know it’s still captcha enabled right ??
